src/runtime/msan.cpp

```diff
@@ -0,0 +1,93 @@
+#include "HalideRuntime.h"
+
+extern "C" {
+
+namespace Halide { namespace Runtime { namespace Internal {
+
+struct Dimension {
+  ptrdiff_t stride_bytes;
+  size_t extent;
+};
+
+WEAK void halide_msan_annotate_buffer_is_initialized_internal(
+    void *user_context, const Dimension *dims, int dims_count, int level, uint8_t* host) {
+
+    const ptrdiff_t stride_bytes = dims[level].stride_bytes;
+    const size_t extent = dims[level].extent;
+    if (level == dims_count - 1) {
+        const size_t swath_bytes = stride_bytes * extent;
+        halide_msan_annotate_memory_is_initialized(user_context, host, swath_bytes);
+        return;
+    }
+
+    for (size_t i = 0; i < extent; ++i) {
+        halide_msan_annotate_buffer_is_initialized_internal(
+            user_context, dims, dims_count, level + 1, host + i * stride_bytes);
+    }
+}
+
+}}} // namespace Halide::Runtime::Internal
+
+WEAK void halide_msan_annotate_memory_is_initialized(void *user_context, void *ptr, size_t len) {
+    // Default implementation does nothing; you must override this and supply (e.g.)
```

Comment:
There is an MSAN call I can declare extern, but it requires the function to really be present at runtime, so if you aren't compiling the rest of the pipeline with MSAN, you'll fail (unless you provide an empty stub). Mainly an issue for the newly added test. Which I could probably fix by compiling the test with MSAN, except of course that not all compilers support it.